controllers/transactionControllers.js

```diff
@@ -8,22 +14,152 @@ export const filterTransactions = async (req, res) => {
   } catch (err) {}
 };
 
-export const updateTransaction = async (req, res) => {
+export const getAllTransactions = async (req, res, next) => {
   try {
-  } catch (err) {}
+    const transactions = await Transaction.find({ user: req.user._id });
+
+    res.json(transactions);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Server error' });
+  }
 };
 
-export const deleteTransaction = async (req, res) => {
+export const createTransaction = async (req, res) => {
+  const { amount, category, date, isIncome, comment } = req.body;
+
+  if (!date || isIncome === undefined)
+    return res.status(400).json({ error: 'Please enter all required information' });
+
+  if (isIncome) delete req.body.category;
+
+  if (amount <= 0) return res.status(400).json({ error: 'The amount must be greather than zero' });
+
+  function convertToDDMMYYYY(dateString) {
```

Comment:
wyrzuć tą funkcję do utils i zaimportuj tutaj i do updateTransaction - naprawi to błąd poniżej i kod będzie czytelniejszy